src/main/scala/rocket/PTW.scala

```diff
@@ -108,14 +108,16 @@ class PTW(n: Int)(implicit edge: TLEdgeOut, p: Parameters) extends CoreModule()(
   val s_ready :: s_req :: s_wait1 :: s_dummy1 :: s_wait2 :: s_wait3 :: s_dummy2 :: s_fragment_superpage :: Nil = Enum(UInt(), 8)
   val state = Reg(init=s_ready)
   val l2_refill_wire = Wire(Bool())
+  val l2_flush_entry_wire = Wire(Bool())
```

Comment:
> Note that the one-entry TLB flush sfence rs1, zero stalls the TLB for one cycle in order to fetch the indexed set and complete the flush. Another (costly) approach would be to flush the indexed set in order to avoid the one-cycle delay, but this would lead to (more) costly page table walks.

In the existing direct-mapped L2TLB, we are doing the other (costly) approach: flushing the indexed set without comparing the tag.
I agree that yours is a worthwhile optimization to make, but I don't want to do that in this same pull request, so I will back out those changes.